packages/grid/_modules_/grid/hooks/features/rows/useGridEditRows.ts

```diff
@@ -84,13 +101,18 @@ export function useGridEditRows(apiRef: GridApiRef) {
         api: apiRef.current,
       });
 
+      const doc = ownerDocument(apiRef.current.rootElementRef!.current as HTMLElement);
+      if (mode === 'view') {
+        doc.removeEventListener('click', handleDocumentClick);
```

Comment:
I would also attach the listener here when changing to the edit mode but by doing that it [broke](https://www.argos-ci.com/mui-org/material-ui-x/builds/1424) the regression tests. The click done to switch between tests is incorrectly caught as a click to exit from the edit mode.